api/services/userService.js

```diff
@@ -0,0 +1,37 @@
+const { userDao } = require('../models')
+const jwt = require('jsonwebtoken')
+
+const { getAccessTokenByKakao, getUserInfoByKakao } = require('./kakaoService')
+
+const kakaoSignIn = async(code) => {
+
+    const kakaoAccessToken = await getAccessTokenByKakao(code)
+    const kakaoUserInfo = await getUserInfoByKakao(kakaoAccessToken)
+
+    const socialId = kakaoUserInfo.sub
+    const userInfo = await userDao.getUserSocialId(socialId)
+
+    if(!userInfo){
+                
+        const newUser = await userDao.signUp(kakaoUserInfo)
+        const payLoad = {userId: newUser.insertId}
+        const accessToken = jwt.sign(
+            payLoad,
+            process.env.JWT_SECRET,
+            { algorithm: process.env.ALGORITHM, expiresIn: process.env.JWT_EXPIRES_IN}
+        )
+        return accessToken   
+
+    } else if (socialId === userInfo.social_id) {
+
+        const payLoad = { userId: userInfo.id }
+        const accessToken = jwt.sign(
+            payLoad,
+            process.env.JWT_SECRET,
+            { algorithm: process.env.ALGORITHM, expiresIn: process.env.JWT_EXPIRES_IN } 
+        )
+        return accessToken
+    }
```

Comment:
작성 잘해주셨네요 재헌님 :) 다만 if, else if에 대해서 다시 생각해보면 좋을 것 같습니다. 어차피 우리 로직은 `이거, 아니면 저거`의 이분법적인 로직인 것이기에 else if가 쓰일 필요가 없고 이미 if 문 안에서 조건문에 걸리지 않는다면 단지 다음 코드로 넘어갈 것 이기때문에 else 조차 필요가 없습니다.
```suggestion
    if(!userInfo){
                
        const newUser = await userDao.signUp(kakaoUserInfo)
        const payLoad = {userId: newUser.insertId}
        const accessToken = jwt.sign(
            payLoad,
            process.env.JWT_SECRET,
            { algorithm: process.env.ALGORITHM, expiresIn: process.env.JWT_EXPIRES_IN}
        )
        return accessToken   
    } 

    const payLoad = { userId: userInfo.id }
    const accessToken = jwt.sign(
        payLoad,
        process.env.JWT_SECRET,
        { algorithm: process.env.ALGORITHM, expiresIn: process.env.JWT_EXPIRES_IN } 
    )
    return accessToken
```